examples/StoreApi/Program.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Momento.Sdk.StoreClient;
+
+namespace Driver;
+
+public class Program
+{
+    public static async Task Main(string[] args)
+    {
+        var client = new StoreClient();
+        await client.CreateStoreAsync("my-store");
+
+        // Can use implicit cast to create StoreValue.
+        await client.PutAsync("my-store", "my-key", "my-value");
+        await client.PutAsync("my-store", "my-key", 42);
+        await client.PutAsync("my-store", "my-key", 3.14);
+        await client.PutAsync("my-store", "my-key", true);
+
+        // The compiler will detect invalid casts.
+        // await client.PutAsync("my-store", "my-key", new string[] { "a", "b" });
+
+        // This demos usage where the developer is sure of the type.
+        {
+            var response = await client.GetAsync("my-store", "string-value");
+            if (response is StoreGet.Success success)
+            {
+                // This uses an implicit cast from StoreValue to string.
+                string value = success.Value;
+                Console.WriteLine($"Success: {value}");
+            }
+            else if (response is StoreGet.Error error)
+            {
+                Console.WriteLine($"Error: {error.Message}");
+            }
+        }
+
+        // A more cautious developer would handle the unfortunate case where the cast fails.
+        {
+            var response = await client.GetAsync("my-store", "bool-value");
+            if (response is StoreGet.Success success)
+            {
+                try
+                {
+                    // This uses an implicit cast from StoreValue to long, but it's actually a bool.
+                    long value = success.Value;
+                    Console.WriteLine($"Success: {value}");
+                }
+                catch (InvalidCastException)

```

Comment:
This one makes me want to raise the same question I raised on your JS PR about whether we should be modeling Success/Error as algebraic types, or just return the value directly and throw an exception for errors. In this case, that would simplify the code down to a try/catch block with multiple catch clauses, as opposed to an if+try.

Probably it'd be too inconsistent to make that dramatic of a change for this Client given that it will be living in the same SDK, but I'm just trying to think through what I would advocate for here if it was a greenfield product.

Reply:
Same comment re: the JS API. I like the shorthand but there's forward compatibility concerns.

Reply:
In what sense? Are you talking about the case where the server changes in the future to support additional response types?

Reply:
Nevermind -- I was confusing the inner value type with the `Success` object. You're saying we can return the `Success` object or bust. We are free to add more data to `Success` if we do that.